lucene/core/src/java/org/apache/lucene/search/similarities/IndriDirichletSimilarity.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * ===============================================================================================
+ * Copyright (c) 2019 Carnegie Mellon University and University of Massachusetts. All Rights
+ * Reserved.
+ *
+ * Use of the Lemur Toolkit for Language Modeling and Information Retrieval is subject to the terms
+ * of the software license set forth in the LICENSE file included with this software, and also
+ * available at http://www.lemurproject.org/license.html
+ *
+ * ================================================================================================
+ */
+package org.apache.lucene.search.similarities;
+
+import java.util.List;
+import java.util.Locale;
+
+import org.apache.lucene.search.Explanation;
+import org.apache.lucene.search.similarities.BasicStats;
+import org.apache.lucene.search.similarities.LMSimilarity;
+
+/**
+ * Bayesian smoothing using Dirichlet priors as implemented in the Indri Search
+ * engine (http://www.lemurproject.org/indri.php). Indri Dirichelet Smoothing!
+ * tf_E + mu*P(t|D) P(t|E)= ------------------------ documentLength + documentMu
+ * mu*P(t|C) + tf_D where P(t|D)= --------------------- doclen + mu
+ */
```

Comment:
Maybe, add a few words giving some intuition about what `mu` does?  It looks like it roughly lets you tune how important document length is in the scoring?

Also, the formatting of the above equations looks like it got garbled?  You will need to use html/javadoc markup to make the formatting survive future developers viewing in browser...

Reply:
I tried adding more formatting and a description of mu.  Let me know if you would like to see anything different or more.  Thanks!